jupyterhub/singleuser/__init__.py

```diff
@@ -63,9 +63,29 @@
             f"Cannot use JUPYTERHUB_SINGLEUSER_EXTENSION={_extension_env} with JUPYTERHUB_SINGLEUSER_APP={_app_env}."
             " Please pick one or the other."
         )
-    from .extension import main
+    try:
+        from .extension import main
+    except ImportError as e:
+        # raise from to preserve original import error
+        raise ImportError(
+            "Failed to import JupyterHub singleuser extension."
+            " Make sure to install dependencies for your single-user server, e.g.\n"
+            "    pip install jupyterlab"
+        ) from e
```

Comment:
I didn't know that this is how one would do something like this, nice!

Reply:
I think this technically not required because it is already the default behavior when raising from within an `except` block, but I think it's nice to be explicit that we want to keep both errors. `raise ... from None` is the inverse - suppressing the original error and only reporting the new one.

Reply:
Ahhh, thanks for helping me learn this!